### PR TITLE
BugFix: Prevent/Restrict the consecutive placeholder based on type

### PR DESCRIPTION
**Pull request by SHIMA0111**
# BugFix: Prevent/Restrict the consecutive placeholder based on type
To avoid oversignt such bugs, extends the test cases, which include corner case generated by AI Agent. 

# Changes generated

- Introduce extensive unit tests for `Template` trait and custom implementations
- Add `validate_template_safety` to prevent ambiguous placeholders in templates
- Enhance parsing with placeholder conflict detection and stricter rules
- Update `templatia-derive` with support for consecutive placeholder constraints
- Expand `templatia-derive` architecture with `utils` and `validator` modules
- Bump version to `0.0.2` across `templatia` and `templatia-derive` packages
- Refactor parser to handle escaped braces and rounding edge cases
- Introduce tests validating consecutive placeholders for `char` and `bool` fields
- Add scenarios for mixed placeholders, literal separators, and edge cases
- Implement tests to ensure correct handling of escaped braces in templates
- Update `templatia-derive` to support consecutive `bool` placeholders
- Integrate `trybuild` for compile-time error testing
- Enhance placeholder parsing with stricter rules and better error handling

## Summary by CodeRabbit

* **New Features**
  * Added template safety validation to prevent unsupported consecutive placeholders while allowing safe consecutive char and bool placeholders.
* **Bug Fixes**
  * Improved error messages and corrected escaped-brace handling and unmatched-brace diagnostics.
* **Tests**
  * Added comprehensive round-trip, edge-case, and compile-fail test suites plus manual Template implementation tests.
* **Chores**
  * Bumped package versions to 0.0.2 and added dev tooling for compile-fail tests.

### CODE REVIEW

File: templatia-derive/src/validator.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+use crate::parser::TemplateSegments;
+use crate::utils::{get_field_type_by_name, get_type_name, is_allowed_consecutive_allowed_type, CONSECUTIVE_PLACEHOLDER_ALLOWED_TYPE};
+
+pub(crate) fn validate_template_safety(
+    segments: &[TemplateSegments],
+    all_fields: &[syn::Field],
+) -> Result<(), String> {
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia-derive/src/validator.rs:7]: This is the crate internal public function so if we describe it, it isn't any effect to the user. 
I think it isn't necessary in this PR.

File: templatia-derive/tests/compile_fail.rs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#[test]
+fn compile_fail_tests()  {
+    let t = trybuild::TestCases::new();
+    t.compile_fail("tests/compile_fail/*.rs");
+
+}
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia-derive/tests/compile_fail.rs:6]: No rust docs needed in test for this current status.

File: templatia/tests/manual_implementations.rs [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+            }
+            
+            let x = parts[0].parse().map_err(|_|
+                TemplateError::Parse("Failed to parse x coordinate".to_string()))?;
+            let y = parts[1].parse().map_err(|_|
+                TemplateError::Parse("Failed to parse y coordinate".to_string()))?;
+            
+            Ok(Point(x, y))
+        }
+    }
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia/tests/manual_implementations.rs:38]: Why? 
This is test. I think we don't need the docs in test. 
The coding guid line indicates the public features' docs should be described. (At least, my think)

File: templatia/tests/manual_implementations.rs [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+                
+            let port_str = parts[1].strip_prefix("port=")
+                .ok_or_else(|| TemplateError::Parse("Missing port= prefix".to_string()))?;
+                
+            let port = port_str.parse::<u16>()
+                .map_err(|_| TemplateError::Parse("Invalid port number".to_string()))?;
+
+            Ok(ServerConfig { name, port })
+        }
+    }
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia/tests/manual_implementations.rs:106]: Why?
This is test. I think we don't need the docs in test.

File: templatia/tests/manual_implementations.rs [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+                return Err(TemplateError::Parse("Expected key=value format".to_string()));
+            }
+
+            let key = parts[0].to_string();
+            let value = parts[1].parse::<T>()
+                .map_err(|e| TemplateError::Parse(format!("Failed to parse value: {}", e)))?;
+
+            Ok(KeyValue { key, value })
+        }
+    }
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia/tests/manual_implementations.rs:168]: Why?
This is test. I think we don't need the docs in test.

File: templatia/tests/manual_implementations.rs [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+                        outer: outer_part.to_string(),
+                        inner,
+                    })
+                } else {
+                    Err(TemplateError::Parse("Invalid inner format, expected [...]".to_string()))
+                }
+            } else {
+                Err(TemplateError::Parse("Expected outer:inner format".to_string()))
+            }
+        }
```
[the remark this answers is not in the document]
Comment (SHIMA0111) [reply templatia/tests/manual_implementations.rs:362]: I've updated it. Please check it.
